src/components/Button/Button.tsx

```diff
@@ -44,8 +45,16 @@ export class Button extends React.Component<IButtonProps, IButtonState> {
                                                // if not utilize default button disabled behavior.
     });
 
+    const iconClassName = css('ms-Icon', iconTypeClass, {
```

Comment:
Problem here is that a primary button will have a white icon when the bg is blue, and a gray icon when the bg is white. In the past I called these "themes" and used some contextual class to trigger the color changes.